### DISCUSSION TITLE
Working with unstructured grids using UXARRAY and xCDAT

### DISCUSSION THREAD

**Discussion by tomvothecoder** [xCDAT/xcdat#382] [3. Ideas]
From our meeting notes (10/11/22)

- The project is still figuring out the best approach to construct their dataset object. I think one of the SEATS goals, is to have xcdat and uxarray complementary to each other. I'm wondering if their design would affect cooperation of both tools. for instance, xcdat to take a regridded-to-latlon dataset from uxarray object, and perform operations under xcdat.. – Jill
  - https://github.com/UXARRAY/uxarray/discussions/154

- Some components of E3SM native output are unstructured
  - atmosphere, land, ocean sea-ice. 
  - Only the river model data is structured lat-lon. 
  - I don't know if there is a technical definition (I need to look for it.). I'm pretty sure Cuvilinear is structured. 
  - The unstructured model output always come with a set of mesh representations. Some modeling centers started to publish unstructured datasets (I can't find the modeling center name (ICON?), but remember that both of us found it when looping over CMIP models)

- Our approach with UXARRAY
  - We probably want to keep things modular and decoupled, so (1) makes more sense. User can open up unstructured datasets in uxarray, then perform operations in xcdat afterwards.
  - Think of the value for an API that opens up an unstructured dataset with uxarray then applies xcdat convenience functions (adding bounds, etc.)

**Comment by tomvothecoder** [comment]
I had question from Forrest Hoffman at the Ninth ESGF Conference about intereropability between UXARRAY and xCDAT.  This means there is probably interest in having this capability one way or another.

**Comment by jypeter** [comment]
This comment is not directly related (my apologies), but having a quick look at this issue reminded me about the following useful document, that I usually recommend to our new students who have grids related questions, and that could possibly be referenced in the **xcdat FAQ** (if somebody wants to mention _model grids_...): [Gridspec: A standard for the description of grids used in Earth System models](https://arxiv.org/abs/1911.08638)

@balaji-gfdl @taylor13 is there something more recent than the paper above?

Learning about [UGRID](https://ugrid-conventions.github.io/ugrid-conventions/), I also checked if it was referenced in the [CF Conventions](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html), but it is [not the case yet](https://github.com/cf-convention/cf-conventions/issues/153)

**Comment by taylor13** [comment]
No, I haven't seen anything more recent regarding Gridspec.   Although Balaji proposed this as a standard, I'm not sure how much buy-in he's gotten.  We should investigate further before considering adopting/implementing anything.
